processing/src/main/java/org/apache/druid/query/aggregation/last/StringLastAggregatorFactory.java

```diff
@@ -149,7 +150,8 @@ public BufferAggregator factorizeBuffered(ColumnSelectorFactory metricFactory)
   @Override
   public boolean canVectorize(ColumnInspector columnInspector)
   {
-    return true;
+    final ColumnCapabilities capabilities = columnInspector.getColumnCapabilities(fieldName);
+    return capabilities != null && capabilities.getType().equals(ValueType.STRING);
```

Comment:
this one probably is necessary to keep though, the string vector aggregator calls `DimensionHandlerUtils.convertObjectToString` which can handle any type of value, however it is using a `VectorObjectSelector` which isn't guaranteed to be implemented for numbers, which typically need to use `VectorValueSelector`.